### DISCUSSION TITLE
Xss preventing livewire 3 from working (array on null error)

### DISCUSSION THREAD

**Discussion by MickProPay** [livewire/livewire#7571] [2. Show and tell]
To anyone having a big project and making use of the following package protonemedia/laravel-xss-protection for xss prevention. This might even be relevant to any xss :P.

So in the protonemedia/laravel-xss-protection said to add it to the Http kernel middleware section so that it catches every request. With livewire 2 its no problem but in livewire 3 it has a full list of routes it now uses to-do all of its fancy livewiry stuff and because of the middleware xss protection, it would create a array on null error because the livewire routes is covered under this middleware protection.

So the fix is to remove it from the kernel and add it to all of your routes individually or create a group and add it so that it excludes the livewire routes.

This problem too me 2.5 weeks to solve and I hope to save someone else that time 👍.

**Comment by joshhanley** [comment]
@MickProPay hmm something doesn't sound right to me as Livewire 2 and Livewire 3 have a similar amount of routes (there is actually less in V3). 

V2 routes:
<img width="1057" alt="image" src="https://github.com/livewire/livewire/assets/882837/0e5d8f7a-1fc4-4c7b-9c86-8645ef23e234">


V3 routes:
<img width="1060" alt="image" src="https://github.com/livewire/livewire/assets/882837/b960ea21-cbe7-4c70-9ee4-fd615622faa1">

So I'm not exactly sure what would have changed to stop it working in V3 if it worked in V2.
